internal/provenance/commit_and_query.go

```diff
@@ -99,39 +97,39 @@ func (s *Store) Commit(blockNum uint64, txsData []*TxDataForProvenance) error {
 		}
 
 		s.logger.Debugf("userID[%s]---(submitted)--->txID[%s]", tx.UserID, tx.TxID)
-		cayleyTx.AddQuad(quad.Make(tx.UserID, SUBMITTED, tx.TxID, ""))
+		batch.WriteQuad(quad.Make(tx.UserID, SUBMITTED, tx.TxID, ""))
 
-		if err := s.addReads(tx, cayleyTx); err != nil {
+		if err := s.addReads(tx, batch); err != nil {
 			return err
 		}
 
-		if err := s.addWrites(tx, cayleyTx, commitSet); err != nil {
+		if err := s.addWrites(tx, batch); err != nil {
 			return err
 		}
 
-		if err := s.addDeletes(tx, cayleyTx); err != nil {
+		if err := s.addDeletes(tx, batch); err != nil {
 			return err
 		}
 	}
 
-	return s.cayleyGraph.ApplyTransaction(cayleyTx)
+	return batch.Close()
 }
 
-func (s *Store) addReads(tx *TxDataForProvenance, cayleyTx *graph.Transaction) error {
+func (s *Store) addReads(tx *TxDataForProvenance, batch graph.BatchWriter) error {
 	for _, read := range tx.Reads {
 		value, err := s.getValueVertex(tx.DBName, read.Key, read.Version)
 		if err != nil {
 			return err
 		}
 
 		s.logger.Debugf("txID[%s]---(reads)--->value[%s]", tx.TxID, quad.NativeOf(value))
-		cayleyTx.AddQuad(quad.Make(tx.TxID, READS, value, ""))
+		batch.WriteQuad(quad.Make(tx.TxID, READS, value, ""))
 	}
 
 	return nil
 }
 
-func (s *Store) addWrites(tx *TxDataForProvenance, cayleyTx *graph.Transaction, commitSet map[string]string) error {
+func (s *Store) addWrites(tx *TxDataForProvenance, batch graph.BatchWriter) error {
```

Comment:
Short question - why we pass to `*graph.Transaction`, but `graph.Batch` - pointer in first case and object itself in second?

Reply:
`graph.BatchWriter` is an interface while `graph.Transaction` is a struct.

Reply:
Thanks.